infrastructure/iam.tf

```diff
@@ -0,0 +1,54 @@
+module "iam_iam-assumable-role-with-oidc" {
+  source = "terraform-aws-modules/iam/aws//modules/iam-assumable-role-with-oidc"
+
+  create_role                    = true
+  role_name                      = "sbraunovic-aws-oidc"
+  provider_url                   = module.iam_iam-github-oidc-provider.url
+  number_of_role_policy_arns     = 1
+  oidc_fully_qualified_audiences = ["sts.amazonaws.com"]
+  role_policy_arns               = [module.iam_iam-policy.arn]
+  oidc_subjects_with_wildcards   = [var.github_repo]
+  tags                           = var.tags
+}
+
+module "iam_iam-github-oidc-provider" {
```

Comment:
As there can be only one github oidc provider per aws account and you are sharing account with Gjovani, please import it into state file if it already exists on the aws side.

Reply:
To do something like this ?
`terraform import aws_iam_openid_connect_provider.this <provider_arn>`

Reply:
Yes, only if you se during plan **phase** that it will be modified which means that it already exists.